### ISSUE TITLE
Badging of app

### ISSUE THREAD

**Issue by ajburls** [dwyl/bestevidence#260]
As the owner of this app, I would like to be able to badge the app by changing colours, logo, "about us" text, and set a flag or field to indicate that the user of the app is using a particular badged version of the app, without the need for involving developers on each occasion it is given to a new professional or health provider organization. 

What resources would be required to achieve this?

**Comment by Cleop** [comment]
First of all we would need to create a definite list of all of the things that would need changing ie. which titles/headers/nav elements etc. would change colour. I seem to remember with the City colours that there were 2 tones, we'd need to agree which elements of design would be colour 1 and which would be colour 2.

We also need to write out which bits of wording would need changing ie. how many times does the org name appear and on what pages, the about page, do you expect to see any changes in the list of options people experience in the sign up process or anywhere else on the app?

Do people manually add the tag that they are from a certain organisation or do you envisage that being logged by the fact that they're accessing the site from a distinct URL?

**Comment by ajburls** [comment]
@Cleop agree with first paragraph, perhaps a dark and a lighter colour?
I think the text to change would be in the about page
May need to change (or add) an organisation specific contact email address - currently messages from top feedback link are coming to me.
Using the badged app needs to automatically tag that this user is from that organisation. I think we want to keep one overall database, not sure what the implications of this are for how it is accessed.

I think we want to create a CMS for us to use to badge apps. Perhaps the first imminent badging as Barts should be done manually and this would then inform the development of the CMS if this isn't adding too much cost?

**Comment by iteles** [comment]
@ajburls That seems like a wise way to go about it.

Do you already know exactly (or even somewhat solidly) what would need to change in each screen of the application for St. Bart's? Both in terms of appearance and in terms of wording?

We should open a separate but linked issue for the automatic tagging of users so that we can keep aesthetics and functionality neatly separated.

**Comment by ajburls** [comment]
No, I think this would need to be decided through discussion with them. I am assuming that the badging would be about altering the appearance of each page (colours and some text) and not the functionality. So, for example where it currently says "BestEvidence" it would say something like "BestEvidence for Barts". The idea would be to keep it simple so that it could be badged for any organization. Currently we have a number of options for selecting on the settings page and it would be useful for this to be able to be added and options added. In terms of Barts, it could be that we would have "Barts user" with this selected and not be able to be unselected but also they many want to say which hospital (Whips Cross, Mile End, Royal London, St Barts or other), So some cost for customisation of this should be included.

The backend database would need to collect these additional features and we need to be able to set up administrative access to the backend database which restricts admin user to only seeing Barts app data.

Barts colours can be seen here: https://www.bartshealth.nhs.uk/

**Comment by iteles** [comment]
## Technical implementation
+ [ ] Create super admin user (no assigning of this role, it's fixed)
+ [ ] Create super admin dash UI + nav item (or reroute)
  + [ ] Create new route
  + [ ] Create new view
  + [ ] Create new page
  + [ ] New auth controller
+ [ ] Creating client functionality
  + [ ] Create client in db
  + [ ] Create client variables - needs further investigation
  + [ ] Store values for text and colours
  + [ ] Use markdown parser
+ [ ] Editing client functionality
+ [ ] Integrate text variables into template to pull in correct text dependent on logged in user
+ [ ] Update About page to pull in text dependent on URL
+ [ ] Add CSS compiler and add functionality to pick up colour hexes from database values to compile
+ [ ] Update CSS files
+ [ ] Add column to users table for institution
+ [ ] Setting the new user's institution based on the URL they sign up from 
+ [ ] Default existing users to required insititution
+ [ ] Create default hexes and text for the current BE app
+ [ ] Implement correct templates dependent on logged in users' institutions


Currently we could add a free input field for the hospital but no additional customisation taken into account here for now.
Note: currently users cannot change institution when they sign up

**Comment by ajburls** [comment]
@iteles Would this  "... each new BE client would get their own URL (e.g. www.bestevidence.info/stbarts or stbarts.bestevidence.info) and the app would automatically recognise and log the new user as part of that institution..." mean there was more than one backend database. (For analytical and automated dashboard views, I would prefer a single backend database with institution as one of the flags that indicated which badged app the data came from.)

**Comment by ajburls** [comment]
@iteles "For now, we are not proposing that the individual clients have dashboards" Why not? This would seem like a very simple thing to do to create a redash user which admin rights over the data flagged as from their organization. I have seen St Barts own dashboard and they draw and integrate data from all sorts of sources and I could imagine them wanting to have access to this data.

**Comment by ajburls** [comment]
@iteles apart from comments above the rest of the proposal seems to be what is required.

**Comment by iteles** [comment]
@ajburls To clarify on your two points above:
+ The data for all St. Bart's users would join our existing data in the central database
+ For now, we are not proposing that the individual clients have dashboards _within the app_, but we would create 2 base queries for them in a Redash dashboard (the ones we already have but specific only to their institution) that they could have access to

**Comment by ajburls** [comment]
@iteles Ines, I think that we need to change the City logo to a Barts logo on the about us page, or at a very minimum be able to get rid of the City logo on apps badged for other organisations.

**Comment by iteles** [comment]
@ajburls Agreed! That's part of the scope, captured within #303
